src/index.tsx

```diff
@@ -166,24 +166,34 @@ const SortableList = <TTag extends keyof JSX.IntrinsicElements = typeof DEFAULT_
       lastTargetIndexRef.current = targetIndex
 
       const isMovingRight = sourceIndex < targetIndex
+      const draggedItemRects = itemsRect.current[sourceIndex]
+      const nextItemRects = itemsRect.current[isMovingRight ? sourceIndex + 1 : sourceIndex - 1]
+
+      if (nextItemRects === undefined) {
+        return
+      }
+
+      const spacingY = isMovingRight
+        ? (nextItemRects?.top ?? 0) - draggedItemRects.bottom
+        : draggedItemRects.top - (nextItemRects?.bottom ?? 0)
+
+      const spacingX = isMovingRight
+        ? (nextItemRects?.left ?? 0) - draggedItemRects.right
+        : draggedItemRects.left - (nextItemRects?.right ?? 0)
+
+      const translateX = (isMovingRight ? -1 : 1) * (draggedItemRects.width + spacingX)
```

Comment:
I will try to do a deeper review over the weekend but I'm sure this will work as expected.
The current code computes the translations for every item independently. Here, you do it only once for the next item. Maybe this works but I have to check more carefully.

Reply:
Yes, I do it for one item only — the dragged one. If we drag an item of height 10, all items should move by its size, not theirs. You can check the storybooks I've created :)